### PR TITLE
don't check in mounter binary

### PR DESCRIPTION
**Pull request by mikedanese**
```release-note
GCI mounter is moved from the manifests tarball to the server tarball.
```

### CODE REVIEW

**Comment by mikedanese** [comment]
So we have a checked in binary and removing it breaks the build...

**Comment by ixdy** [comment]
what is using it?

**Comment by ixdy** [comment]
it looks like `mounter` was a shell script until #41944.

assuming the mounter binary is the output of `mounter.go`, we should probably integrate this into the build process, or move it out-of-tree...

cc @jingxu97 @vishh @msau42

**Comment by spxtr** [comment]
That's nice.

**Comment by ixdy** [comment]
incidentally, https://github.com/kubernetes/kubernetes/pull/48969 will start building the mounter binary every time for bazel builds.

**Comment by spxtr** [comment]
Please rebase.

**Comment by spxtr** [comment]
Can we check this in? It looks good to me but I feel bad putting on LGTM because I don't have any idea what the mounter binary even does.

**Comment by msau42** [comment]
What are the etcd changes? Are they related to the mounter binary?

Please run the 3 "GCP Volumes" test on GCE using GCI.  Those tests will test this mounter code path.

**Comment by jingxu97** [comment]
mounter binary is for containerized mount feature we added earlier. The new COS does not have the nfs/gluster mount binary installed so mount operation will fail. Currently we installed the mount binary under /home/kubernetes/containerized_mount and mounter binary use chroot to the mount directory and then mount. As @msau42 before checking in the code, we need to make sure the Volumes tests pass.

File: cluster/gce/BUILD [hunk 1]
```diff
@@ -9,7 +9,6 @@ pkg_tar(
         "container-linux/configure-helper.sh",
         "gci/configure-helper.sh",
         "gci/health-monitor.sh",
-        "gci/mounter/mounter",
```
Comment (ixdy) [inline cluster/gce/BUILD]: remove the symlink too?

File: cluster/gce/gci/mounter/BUILD [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+licenses(["notice"])
+
+load(
+    "@io_bazel_rules_go//go:def.bzl",
+    "go_binary",
+    "go_library",
+)
+
+go_binary(
+    name = "mounter",
```
Comment (ixdy) [inline cluster/gce/gci/mounter/BUILD]: this is currently statically linked - is it still statically linked when built each time?

**Comment by ixdy** [comment]
@mikedanese want to rebase this?

**Comment by mikedanese** [comment]
@ixdy rebased.

**Comment by ixdy** [comment]
LGTM, though you can probably squash the commits, since they don't really make sense as separate pieces anymore.

**Comment by ixdy** [comment]
Test failure looks legitimate:
```
I1026 18:46:41.613] unable to create test archive: failed to setup test package "/tmp/node-e2e-archive031296679": Could not find GCI/COS mounter script at "/go/src/k8s.io/kubernetes/cluster/gce/gci/mounter/mounter"! If this script has been (re)moved, please update the e2e node remote test runner accordingly! Err: stat /go/src/k8s.io/kubernetes/cluster/gce/gci/mounter/mounter: no such file or directory.
```

File: test/e2e_node/builder/build.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ var buildTargets = []string{
 	"cmd/kubelet",
```
Comment (ixdy) [inline test/e2e_node/builder/build.go]: it makes me sad that these are not just bazel dependencies.

File: test/e2e_node/remote/node_e2e.go [hunk 4]
```diff
@@ -110,51 +104,11 @@ func tarAddFile(tar, source, dest string) error {
 	return nil
 }
 
-// Includes the GCI/COS mounter artifacts in the deployed tarball
```
Comment (ixdy) [inline test/e2e_node/remote/node_e2e.go]: so much less code. yes!

**Comment by ixdy** [comment]
Do you think it warrants a release note that the gci mounter binary was moved from the manifests tarball to the server tarball?

**Comment by mikedanese** [comment]
Added a release note.

/approve no-issue

**Pull request merged**